src/js/actions/layers.js

```diff
@@ -1528,6 +1529,37 @@ define(function (require, exports) {
     editVectorMask.writes = [locks.PS_DOC];
     editVectorMask.transfers = [tools.select];
 
+    var addVectorMask = function () {
+        var applicationStore = this.flux.store("application"),
+            currentDocument = applicationStore.getCurrentDocument(),
+            toolStore = this.flux.store("tool"),
+            superselectVector = toolStore.getToolByID("superselectVector");
+
+        if (currentDocument !== null) {
```

Comment:
If the current Document is null, this action will cause a throw, because it doesn't return a promise.